src/app.ts

```diff
@@ -141,38 +159,38 @@ for (const file of monitorFiles) {
 // Load other client events
 require('./events');
 
-client.on('ready', async () => {
-	if (!client || !client.user) throw new Error(`Bot not logged in and ready event triggered.`); // Should never happen
+client.on('ready', () => void (async () => {
+	if (!client?.user) throw new Error(`Bot not logged in and ready event triggered.`); // Should never happen
 	console.log(`Logged in as ${client.user.tag}.`);
 
 	// Startup events
 
 	await Promise.all([
-		...commands.map(cmd => {
-			// Aliases cannot be initalized so we skip them
-			if (typeof cmd === 'function') cmd.init();
-		}),
+		...commands
+			// Aliases cannot be initialized so we skip them
+			.filter(cmd => typeof cmd === 'function')
+			.map(cmd => (cmd as Constructable<BaseCommand>).init()),
 		...monitors.map(monitor => monitor.init()),
 	]);
-});
+})());
```

Comment:
This is an event listener, iirc this change here calls the function after creation? Correct me if I'm wrong its been years since I've seen this syntax.

Reply:
Yes, I had to wrap your async callback in an async function and then call it and mark it with void.